jit-service/src/main/java/org/dcsa/jit/service/TimestampService.java

```diff
@@ -229,4 +233,20 @@ private void ensureValidUnLocationCode(String unLocationCode) {
               + " not part of reference implementation data set");
     }
   }
+
+  private Facility ensureValidFacilityCode(LocationTO locationTO) {
+    if (locationTO.UNLocationCode() != null && locationTO.facilityCode() != null) {
+      return facilityRepository
+        .findByUNLocationCodeAndFacilitySMDGCode(
+          locationTO.UNLocationCode(), locationTO.facilityCode())
+        .orElseThrow(
```

Comment:
This is not correct if the location has a `BIC` code

Reply:
I agree, will fix that

Reply:
The DCSA-Event-Core (from the R2DBC days) had a `FacilityService.findByUNLocationCodeAndFacilityCode` which would do the lookup correctly given the UNLocationCode, the facilityCodeListProvider and the facilityCode. I think it would make sense to revive that service/implementation

Reply:
I will look into it, thanks for the tip.